src/client-functions/selector-builder/index.js

```diff
@@ -144,6 +146,22 @@ export default class SelectorBuilder extends ClientFunctionBuilder {
         if (snapshot && !this.options.counterMode) {
             this._addBoundArgsSelectorGetter(snapshot, selectorArgs);
             createSnapshotMethods(snapshot);
+
+            if (this.options.customMethods) {
```

Comment:
I think we should add custom methods to snapshot, at least for consistency. Besides, are there any performance problem with it?